libafl/src/schedulers/mod.rs

```diff
@@ -104,16 +114,38 @@ impl<S> Scheduler for RandScheduler<S>
 where
     S: HasCorpus + HasRand,
 {
+    fn on_add(&mut self, state: &mut Self::State, idx: CorpusId) -> Result<(), Error> {
+        // Set parent id
+        let current_idx = *state.corpus().current();
+        state
+            .corpus()
+            .get(idx)?
+            .borrow_mut()
+            .set_parent_id_optional(current_idx);
```

Comment:
@domenukk 
setting parent id here.

Reply:
Is this even called for Objectives?

Reply:
No. i put set_parent_id_optional back for objectives

Reply:
schedulers change the data of testcases when on_add, on_next... happens. so I think it's better to put them together.